src/PowerShellEditorServices.Protocol/Server/LanguageServerSettings.cs

```diff
@@ -66,31 +66,42 @@ public ScriptAnalysisSettings()
 
                 string settingsPath = settings.SettingsPath;
 
-                if (string.IsNullOrWhiteSpace(settingsPath))
+                try
                 {
-                    settingsPath = null;
-                }
-                else if (!Path.IsPathRooted(settingsPath))
-                {
-                    if (string.IsNullOrEmpty(workspaceRootPath))
+                    if (string.IsNullOrWhiteSpace(settingsPath))
                     {
-                        // The workspace root path could be an empty string
-                        // when the user has opened a PowerShell script file
-                        // without opening an entire folder (workspace) first.
-                        // In this case we should just log an error and let
-                        // the specified settings path go through even though
-                        // it will fail to load.
-                        logger.Write(
-                            LogLevel.Error,
-                            "Could not resolve Script Analyzer settings path due to null or empty workspaceRootPath.");
+                        settingsPath = null;
                     }
-                    else
+                    else if (!Path.IsPathRooted(settingsPath))
                     {
-                        settingsPath = Path.GetFullPath(Path.Combine(workspaceRootPath, settingsPath));
+                        if (string.IsNullOrEmpty(workspaceRootPath))
+                        {
+                            // The workspace root path could be an empty string
+                            // when the user has opened a PowerShell script file
+                            // without opening an entire folder (workspace) first.
+                            // In this case we should just log an error and let
+                            // the specified settings path go through even though
+                            // it will fail to load.
+                            logger.Write(
+                                LogLevel.Error,
+                                "Could not resolve Script Analyzer settings path due to null or empty workspaceRootPath.");
+                        }
+                        else
+                        {
+                            settingsPath = Path.GetFullPath(Path.Combine(workspaceRootPath, settingsPath));
+                        }
                     }
                 }
+                catch (Exception ex) when (ex is NotSupportedException)
+                {
+                    // Invalid chars in path like ${env:HOME} can cause Path.GetFullPath() to throw, catch such errors here
+                    logger.WriteException(
+                        $"Invalid Script Analyzer settings path - '{settingsPath}'.",
+                        ex);
+                }
 
                 this.SettingsPath = settingsPath;
```

Comment:
Do we still want to set `this.SettingsPath` to `settingsPath` even if `settingsPath` is an Invalid Script Analyzer settings path?

Reply:
Doh!  No.  Thanks for catching this.  I'll fix it later tonight.

Reply:
Fixed.